container-engine-lib/lib/backend_impls/docker/docker_manager/docker_auth.go

```diff
@@ -0,0 +1,183 @@
+
+	"github.com/docker/docker/api/types/registry"
+	dockerregistry "github.com/docker/docker/registry"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+const (
+	ENV_DOCKER_CONFIG string = "DOCKER_CONFIG"
+)
+
+// RegistryAuthConfig holds authentication configuration for a container registry
+type RegistryAuthConfig struct {
+	Auths       map[string]registry.AuthConfig `json:"auths"`
+	CredHelpers map[string]string              `json:"credHelpers"`
+	CredsStore  string                         `json:"credsStore"`
+}
+
+// loadDockerAuth loads the authentication configuration from the config.json file located in $DOCKER_CONFIG or ~/.docker
+func loadDockerAuth() (RegistryAuthConfig, error) {
+	configFilePath := os.Getenv(ENV_DOCKER_CONFIG)
+	if configFilePath == "" {
+		configFilePath = os.Getenv("HOME") + "/.docker/config.json"
+	} else {
+		configFilePath = configFilePath + "/config.json"
+	}
+
+	file, err := os.ReadFile(configFilePath)
+	if err != nil {
+		return RegistryAuthConfig{}, stacktrace.Propagate(err, "error reading Docker config file")
+	}
+
+	var authConfig RegistryAuthConfig
+	if err := json.Unmarshal(file, &authConfig); err != nil {
+		return RegistryAuthConfig{}, stacktrace.Propagate(err, "error unmarshalling Docker config file")
+	}
+
+	return authConfig, nil
+}
+
+// getRegistriesFromCredsStore fetches all registries from a Docker credential helper (credStore)
+func getRegistriesFromCredsStore(credHelper string) ([]string, error) {
+	credHelperCmd := "docker-credential-" + credHelper
+
+	cmd := exec.Command(credHelperCmd, "list")
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return nil, stacktrace.Propagate(err, "error executing credential helper %s: %s", credHelperCmd, stderr.String())
+	}
+	// Output will look like this: {"https://index.docker.io/v1/":"username"}
+	var result map[string]string
+	outStr := out.String()
+	err := json.Unmarshal([]byte(outStr), &result)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "error unmarshaling credential helper list output %s: %s", credHelperCmd, outStr)
+	}
+
+	registries := []string{}
+	for k := range result {
+		registries = append(registries, k)
+	}
+	return registries, nil
+}
+
+// getCredentialsFromStore fetches credentials from a Docker credential helper (credStore)
+func getCredentialsFromStore(credHelper string, registryURL string) (*registry.AuthConfig, error) {
+	// Prepare the helper command (docker-credential-<store>)
+	credHelperCmd := "docker-credential-" + credHelper
+
+	// Execute the credential helper to get credentials for the registry
+	cmd := exec.Command(credHelperCmd, "get")
+	cmd.Stdin = strings.NewReader(registryURL)
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return nil, stacktrace.Propagate(err, "error executing credential helper %s: %s", credHelperCmd, stderr.String())
```

Comment:
I missed this but can we change `credHelperCmd` to `cmd.String()` this way an error outputs `docker-credential-<store> get` or `.. list` as opposed to just `docker-credential-<store>`

Reply:
went ahead and made the fix!